src/libs/Navigation/AppNavigator/createCustomStackNavigator/index.tsx

```diff
@@ -61,20 +59,25 @@ function ResponsiveStackNavigator(props: ResponsiveStackNavigatorProps) {
     const {stateToRender, searchRoute} = useMemo(() => {
         const routes = reduceCentralPaneRoutes(state.routes);
 
-        const lastRoute = routes[routes.length - 1];
-        const isLastRouteSearchRoute = getTopmostCentralPaneRoute({routes: [lastRoute]} as State<RootStackParamList>)?.name === SCREENS.SEARCH.CENTRAL_PANE;
-
-        const firstRoute = routes[0];
-
-        // On narrow layout, if we are on /search route we want to hide all central pane routes and show only the bottom tab navigator.
-        if (isSmallScreenWidth && isLastRouteSearchRoute) {
+        // On narrow layout, if we are on /search route we want to hide the search central pane route.
+        if (isSmallScreenWidth) {
+            let lastSearchCentralPane: RouteProp<ParamListBase> | undefined;
+            const filteredRoutes = routes.filter((route) => {
+                const isSearchCentralPane =
+                    route.name === NAVIGATORS.CENTRAL_PANE_NAVIGATOR &&
+                    ((!!route.params && 'screen' in route.params && route.params.screen === SCREENS.SEARCH.CENTRAL_PANE) || route.state?.routes.at(-1)?.name === SCREENS.SEARCH.CENTRAL_PANE);
+                if (isSearchCentralPane) {
+                    lastSearchCentralPane = route;
+                }
+                return !isSearchCentralPane;
+            });
```

Comment:
Instead of using both findLast and filter, I only use filter so we only loop once.

Reply:
Oh wait, I actually simplify it a bit more based on the previous code